File: Remove Nth Node From End of List - Leetcode 19/Remove Nth Node From End of List - Leetcode 19.java
class ListNode {
    int val;
    ListNode next;

    ListNode(){}
    ListNode(int val){
        this.val = val;
    }
    ListNode(int val, ListNode next){
        this.val = val;
        this.next = next;
    }
}

class Solution {
    public ListNode removeNthFromEnd(ListNode head)
    {
        ListNode d = new ListNode(0);
        d.next = head;
        ListNode a = d, b = d;
        int c = 0;

        while (c < n+1 && a != null )
        {
            a = a.next;
            c++;
        }

        while (a != null && b != null)
        {
            a = a.next;
            b = b.next;
        }

        b.next = b.next.next;

        return d.next;
    }
}

File: Remove Nth Node From End of List - Leetcode 19/Remove Nth Node From End of List - Leetcode 19.py
class ListNode:
    def __init__(self, val = 0, next = None):
        self.val = val;
        self.next = next;

def removeNthFromEnd(head:ListNode)->ListNode:
    d = ListNode()
    d.next = head
    a,b = d,d
    c = 0

    for _ in range(n + 1):
        a = a.next
    
    while a and b:
        a = a.next
        b = b.next

    b.next = b.next.next

    return d.next
